src/node_api.cc

```diff
@@ -2197,6 +2252,68 @@ napi_status napi_get_value_int64(napi_env env,
   return napi_clear_last_error(env);
 }
 
+napi_status napi_get_value_bigint_int64(napi_env env,
+                                        napi_value value,
+                                        int64_t* result,
+                                        bool* lossless) {
```

Comment:
@nodejs/n-api Side note … did we make a conscious decision to allow `bool` in our API? That makes N-API incompatible with older C versions, which is a shame … and this [does not seem to be just theoretical](https://github.com/libuv/libuv/pull/1885) :slightly_frowning_face:

Reply:
```cc
int size2int(size_t val) {
  return (val <= INT_MAX) ? (int)((ssize_t) val) : -1;
}

size_t int2size_t(int val) {
    return (val < 0) ? __SIZE_MAX__ : (size_t)((unsigned)val);
}
```

Reply:
would something like this work for you?

```cc
int size2int(size_t n) {
  if (val <= INT_MAX) {
    return (int)((ssize_t) n);
  }
  return -1;
}

size_t int2size(int n) {
  if (val < 0) {
    return SIZE_MAX;
  return (val < 0) ? __SIZE_MAX__ : (size_t)((unsigned)val);
}
```

Reply:
what did C use before `bool`?

Reply:
@devsnek Any integral type, and macros representing true (usually 1) and false (0).